dev-tools/mage/dockerbuilder.go

```diff
@@ -189,11 +194,15 @@ func (b *dockerBuilder) expandDockerfile(templatesDir string, data map[string]in
 	return nil
 }
 
-func (b *dockerBuilder) dockerBuild() (string, error) {
+func (b *dockerBuilder) dockerBuild(variant string) (string, error) {
 	tag := fmt.Sprintf("%s:%s", b.imageName, b.Version)
+	if variant != "" {
```

Comment:
Would it be worth it to check if the variant is present in `b.PackageSpec.Variants`? I know that in https://github.com/elastic/beats/pull/27052/files#diff-2d000e355fe01e44a412b7e21904c617cc0953e8da569f66608d40068c08f163R70 the variants are collected from the `PackageSpec`, but what would happen if this method is called with a random and non-existing string? If that is not allowed, as a non-exposed API, then it's OK as it is, although that defensive check would cause no harm.

Same thing for the `prepareBuild(variant)` method

Reply:
`""` is the default variant. If it's called with another string it should work fine allowing for use of other conditionals.